packages/flutter_test/lib/src/accessibility.dart

```diff
@@ -312,48 +312,138 @@ class MinimumTextContrastGuideline extends AccessibilityGuideline {
+
+  /// The [Finder] used to find a subset of elements.
+  ///
+  /// [finder] determines which subset of elements will be tested for
+  /// contrast ratio.
+  final Finder finder;
+
+  final String _description;
+
+  @override
+  String get description => _description;
+
+  @override
+  Future<Evaluation> evaluate(WidgetTester tester) async {
+    // Compute elements to be evaluated.
+
+    final List<Element> elements = finder.evaluate().toList();
+
+    // Obtain rendered image.
+
+    final RenderView renderView = tester.binding.renderView;
+    final OffsetLayer layer = renderView.debugLayer as OffsetLayer;
+    ui.Image image;
+    final ByteData byteData = await tester.binding.runAsync<ByteData>(() async {
+      // Needs to be the same pixel ratio otherwise our dimensions won't match the
+      // last transform layer.
+      image = await layer.toImage(renderView.paintBounds, pixelRatio: 1 / tester.binding.window.devicePixelRatio);
+      return image.toByteData();
+    });
+
+    // How to evaluate a single element.
+
+    Evaluation evaluateElement(Element element) {
+      final RenderBox renderObject = element.renderObject as RenderBox;
+
+      final Rect originalPaintBounds = renderObject.paintBounds;
+
+      final Rect inflatedPaintBounds = originalPaintBounds.inflate(4.0);
+
+      final Rect paintBounds = Rect.fromPoints(
+        renderObject.localToGlobal(inflatedPaintBounds.topLeft),
+        renderObject.localToGlobal(inflatedPaintBounds.bottomRight),
+      );
+
+      final List<int> subset = _subsetFromRect(byteData, paintBounds, image.width, image.height);
+
+      if (subset.isEmpty) {
+        return const Evaluation.pass();
+      }
+
+      final ContrastReport report = ContrastReport(subset);
+      final double contrastRatio = report.contrastRatio();
+
+      if (contrastRatio >= minimumRatio - tolerance) {
+        return const Evaluation.pass();
+      } else {
+        return Evaluation.fail(
+            '$element:\nExpected contrast ratio of at least '
+                '$minimumRatio but found ${contrastRatio.toStringAsFixed(2)} \n'
+                'The computed light color was: ${report.lightColor}, '
+                'The computed dark color was: ${report.darkColor}\n'
+                '$description'
+        );
       }
     }
-    return buffer;
-  }
 
-  @override
-  String get description => 'Text contrast should follow WCAG guidelines';
+    // Collate all evaluations into a final evaluation, then return.
+
+    Evaluation result = const Evaluation.pass();
+
+    for (final Element element in elements) {
+      result = result + evaluateElement(element);
+    }
+
+    return result;
+  }
 }
 
-class _ContrastReport {
-  factory _ContrastReport(List<int> colors) {
+/// A class that reports the contrast ratio of a part of the screen.
+///
+/// Commonly used in accessibility testing to obtain the contrast ratio of
+/// text widgets and other types of widgets.
+class ContrastReport {
```

Comment:
I like the documentation added, but would prefer that this class stay private.

Reply:
Done.